book/src/04_3_1_i2c.md

```diff
@@ -1,21 +1,132 @@
+| Temperature and Humidity  | SHTC3        | [Datasheet](https://www.sensirion.com/en/environmental-sensors/humidity-sensors/digital-humidity-sensor-shtc3-our-new-standard-for-consumer-electronics/)            | [Link](https://crates.io/crates/shtcx)       | 0x70 |
 
-* Define two pins, one as SDA and one as SCL. Use them to instantiate an Instance of the I²C peripheral. 
 
-* Using the existing driver crates, make an instance of each sensor, passing the I²C instance into them. 
+ The task is to use an existing driver from crates.io to read out the temperature and humidity sensor over i2c. After that, a second sensor will be read out over the same i2c bus using `shared-bus`. The driver for the second sensor is available locally in `common/`.
 
+### Part 1: Reading Temperature & Humidity
 
-TODO `i2c-driver/exercise/src/main.rs` contains working code. Running it without modifying the driver file will yield errors. 
+Create an instance of the of the Humidity sensor SHTC3 and read and print the values for humidity and temperature every 600 ms.
 
-`i2c-driver/solution/src/imc42670p.rs` provides a solution to the task. 
 
 
+`i2c-sensor-reading/examples/part_1.rs` contains a working solution of Part 1.
 
+`i2c-sensor-reading/src/main.rs` contains skeleton code, that already contains necessary imports for this part. 
 
+#### Steps:
 
+✅ Go to the `i2c-sensor-reading/` folder and open the relevant documentation with the following command:
+  
+```
+$ cargo doc --open
+```
+ 
+✅ Define two pins, one as SDA and one as SCL. 
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+✅ Create an Instance of the I²C peripheral with the help of the documentation. 
+
+✅ Use the `shtcx` driver crate, make an instance of the SHTC3 sensor passing the I²C instance into them. Check the documentation for guidance. 
+
+✅ To check if the sensor is addressed correctly, read it's device ID and print the value.
+
+#### Expected Output:
+```
+Device ID: 71
+```
+
+✅ Make a measurement, and read the sensor values and print them. Check the documentation for guidance on sensor methods. 
+
+#### Expected Output:
+
+```
+TEMP: [local temperature] °C
+HUM: [local humidity] %
+```
+
+❗ Some sensors need some time to pass between measurement and reading value. 
+❗ Watch out for the expected units!
+
+
+
+
+
+## Hints
+
+* There are methods that turn the sensor values into the desired unit. 
+
+### Part 2: Reading Accelerometer data. 
+
+Using a bus manager, implement the second sensor. Read out its values and print the values from both sensors. 
+
+
+Continue with your code from Part 1. Alternatively you can start with the provided partial solution of Part 1: `i2c-sensor-reading/examples/part_1.rs`.
+
+`i2c-sensor-reading/examples/part_2.rs` contains a working solution of Part 1.` contains a working solution of Part 2. You can consult it if you need help. 
```

Comment:
Seems like these sentences are not correct / either the first or second sentence should be removed.

Reply:
Ah yes I understand what you mean.